databroker/core.py

```diff
@@ -1218,12 +1234,19 @@ def __init__(self,
 
         self._run_stop_doc = metadata['stop']
         self._run_start_doc = metadata['start']
-        self._descriptors = [descriptor for descriptor in metadata['descriptors']
-                             if descriptor.get('name') == self._stream_name]
-        # Should figure out a way so that self._resources doesn't have to be
-        # all of the Run's resources.
-        self._resources = metadata['resources']
         self._partitions = None
+        self._load()
```

Comment:
Note to selves: we think this shouldn't be necessary. Make sure it's removed (or understood) before merging.

Reply:
updated